### PR TITLE
Pap 218 frontend implement data persistence and functionality

### PR DESCRIPTION
**Pull request by bobbygrdn**
## Ticket

ID:  PAP-221
Link: https://project-ascend-io.atlassian.net/browse/PAP-218?atlOrigin=eyJpIjoiNmJmNDA2YTU4ZjRiNDJhMmJlNTZmOTI0M2RkODdhNWQiLCJwIjoiaiJ9
Design File: https://www.figma.com/design/siiCloyWGa7e19lYRr8RP4/Messanging?node-id=0-1&node-type=canvas&t=3M0OEULDR89OcSz6-0

## Problem

The application needs a real-time direct messaging feature so users can converse with one another. The frontend needed updated logic to handle functional requirements and data persistence.

## Solution

Phase 2: I updated the logic to encompass the functional and data persistence requirement according to the design file. Web Socket functionality and real-time updates will be implemented in Phase 3.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Manual testing
Proper testing will be conducted during phase 4.

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Remarks

If anything seems too complex or doesn't make sense, please reach out so we can discuss its purpose or update the logic for better understanding.

### CODE REVIEW

**Review by diakymenko — COMMENTED** [review]
Hi Robert!

These are really fundamental changes—great job taking it on!

As far as I understand, direct messaging functionality is out of the scope of this PR, so I’m only reviewing the design changes and updates to existing features.

A couple of issues I encountered:

1. The error on initial app launch:
ERROR
Server error! status: 404
    at eval (webpack-internal:///./src/services/conversationsService.ts:17:19)
    at Generator.next (<anonymous>)
    at fulfilled (webpack-internal:///./src/services/conversationsService.ts:5:58)

2. The compilation time seems to have increased compared to what we had earlier. I’m not sure if it’s related to the error above.
3. I got the same error as above while logging in.
4. Not a blocker. A small UI improvement suggestion for the Settings drop-down: Should we have a visual state for Email Settings being selected/unselected? Right now, when we click on Email Config settings, it collapses back into Settings. We can discuss possible options with Kaitlyn.

File: src/Components/sidebar/Sidebar.tsx
Comment (Amenu-sh) [inline src/Components/sidebar/Sidebar.tsx]: Navigate("/settings");  // this fixes displaying current configurations when clicking on settings

**Review by Amenu-sh — COMMENTED** [review]
Great work, Robert! I really liked the new UI, especially the responsiveness, navigation, and flow—particularly in the Messages section, which looks impressive. However, it seems that everything is UI-focused for now and not yet functional in real-time.

I noticed one issue: when clicking items in the sidebar, the view remains on Messages unless we first click Settings and then select Email Configurations to view the current configuration.

I have two suggestions:

1. We could simplify by removing the sub-list under Settings and updating the onClick function to directly display the configuration.
const handleSettingsClick = (): void => {
    setActiveItem("settings");
    Navigate("/settings");
    // Remove: setShowSettingsList(!showSettingsList);
};
2. On the main (right) side when clicking settings, display nothing initially but show the relevant components when the list (Email-configuration) under Settings is clicked.
These are just my thoughts—it's important for the whole team, especially the UI designers, to agree on the best approach.

**Comment by bobbygrdn** [comment]
This PR is stage 2. So stage 1 PR needs to be reviewed and changed if needed first. This PR feeds off of the UI changes of that one.

**Comment by bobbygrdn** [comment]
Thanks for reviewing @diakymenko. 

I assume that the error is caused by you using the main branch of the backend. This PR will need to be run in conjunction with the PAP-9 branch of the backend because it adds the backend logic for these changes. Let me know if that fixes the error you are getting.

I will update you on when I need further review after changes/updates from the rest of the team.

File: src/Components/sidebar/Sidebar.tsx
Comment (bobbygrdn) [reply src/Components/sidebar/Sidebar.tsx]: Thanks for reviewing Amenu. You are correct that this does not implement real-time functionality yet—just functional logic and data persistence. Once I hear back from the design team I will consider your solution and adapt it to their guidance.

Comment (bobbygrdn) [reply src/Components/sidebar/Sidebar.tsx]: The design team has pushed this to the next sprint so I will resolve this conversation. I think we should keep your idea in mind during the next sprint. It is a good solution.

**Comment by diakymenko** [comment]
@bobbygrdn, thanks for the heads up! I was able to run the app with the right backend branch and it works as expected. Great job on UI changes!

**Review by diakymenko — DISMISSED** [review]

**Comment by bobbygrdn** [comment]
@diakymenko I am not sure why it says I dismissed your approval. I didn't touch anything. Haha. Can you approve again please? Thank you!

**Review by diakymenko — DISMISSED** [review]
Approved

**Comment by bobbygrdn** [comment]
@diakymenko I'm not sure what's happening. One more time please. 😁

**Review by diakymenko — DISMISSED** [review]

**Comment by diakymenko** [comment]
@bobbygrdn I can see this msg after I submit my approval: `The merge-base changed after approval.`. Could it be that main branch has changed?

**Comment by bobbygrdn** [comment]
@diakymenko Hmm. Okay, something else is going on. I'll look into it tomorrow. Thanks for reviewing. 😁

Possibly, I am not sure. So strange.

**Comment by bobbygrdn** [comment]
@diakymenko @Amenu-sh, I believe that should fix the merge base issue we had before. I just need one final review, and then I will merge. Thanks!

**Review by diakymenko — DISMISSED** [review]

**Review by diakymenko — APPROVED** [review]

**Pull request merged**
